raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -1028,39 +1060,31 @@ contract TokenNetwork is Utils {
     }
 
     /// @dev Returns the channel specific data.
-    /// @param participant1 Address of one of the channel participants.
-    /// @param participant2 Address of the other channel participant.
-    /// @return Channel state and settle_block_number.
-    function getChannelInfo(address participant1, address participant2)
+    /// @param channel_identifier Identifier for the channel on which this operation takes place.
+    /// @return Channel settle_block_number and state.
+    function getChannelInfo(uint256 channel_identifier)
         view
         external
-        returns (uint256, uint256, uint8)
+        returns (uint256, uint8)
     {
-        uint256 channel_identifier;
-
-        channel_identifier = getChannelIdentifier(participant1, participant2);
         Channel storage channel = channels[channel_identifier];
 
         return (
-            channel_identifier,
             channel.settle_block_number,
             channel.state
```

Comment:
We can have a state for settled channels, if `channel.state == 0 && channel_identifier <= channel_counter` the state settled can be returned.

Reply:
Agreed and makes sense. I will use a `uint8` for this in the current PR. It can be changed to an `enum` state with https://github.com/raiden-network/raiden-contracts/pull/206.

For the record, we still have `settled_but_pending_unlock` not covered. It can be covered if the changes from https://github.com/raiden-network/raiden-contracts/pull/203 are merged, but it would require additional arguments in the `getChannelInfo` getter - e.g. `function getChannelInfo(uint256 channel_identifier, address participant1, address participant2)`, so the smart contract can check the stored `unlock` data. We can discuss this later I supposed, after this PR and the one above are clarified.

Reply:
+1.

With the check augusto suggests we can have a clear view of the channel state from the raiden client side using `getChannelInfo()`

Reply:
Added in https://github.com/raiden-network/raiden-contracts/pull/198/commits/388ff43356f619827691513ee12b301d95a4ab9f